islandora_datastream_crud.drush.inc

```diff
@@ -345,17 +345,20 @@ function drush_islandora_datastream_crud_fetch_pids() {
       dt('Sorry, no PIDS were found.'));
   }
   else {
+    module_load_include('inc', 'islandora', 'includes/utilities');
     foreach ($pids as $pid) {
-      if (drush_get_option('pid_file')) {
-        file_put_contents(drush_get_option('pid_file'), $pid . PHP_EOL, FILE_APPEND);
-      }
-      else {
-        drush_print($pid);
+      if (islandora_namespace_accessible($pid)) {
+        if (drush_get_option('pid_file')) {
+          file_put_contents(drush_get_option('pid_file'), $pid . PHP_EOL, FILE_APPEND);
+        }
+        else {
+          drush_print($pid);
+        }
+        if (drush_get_option('pid_file')) {
+          drush_log(dt("PIDS written to file !pid_file", array('!pid_file' => drush_get_option('pid_file'))), 'ok');
+        }
       }
     }
-    if (drush_get_option('pid_file')) {
-      drush_log(dt("PIDS written to file !pid_file", array('!pid_file' => drush_get_option('pid_file'))), 'ok');
-    }
   }
```

Comment:
> "PIDs written to file" repeats for every result:

If you put the drush_log block back where it was in the original, it only prints once.